### PR TITLE
Added fade helpers to help clean code in game objects

### PR DESCRIPTION
**Pull request by Semphriss**
The FadeHelper will make it much easier to write fading code inside GameObjects and any `update()`-able class.

**Note:** This helper has yet to be used inside the code. A refactor would be needed. I can do this, but if you think this PR is a good idea, I would really appreciate your help :)

### CODE REVIEW

File: src/util/fade_helper.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                        m_start + (m_target - m_start) * (m_time / m_total_time)
+                        )));
+    if (m_value)
+      *m_value = progress;
+    return progress;
+  }
+}
+
+bool
+FadeHelper::completed()
```
Comment (tobbi) [inline src/util/fade_helper.cpp:69]: const.

File: src/util/fade_helper.hpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  /**
+   * Increases the internal timer of the FadeHelper.
+   * 
+   * @param dt_sec The increase in time since last call to update().
+   * @returns The new value for the internal value.
+   */
+  float update(float dt_sec);
+
+  /** @returns true if the FadeHelper has completed fading. */
+  bool completed();
```
Comment (tobbi) [inline src/util/fade_helper.hpp:43]: And here too.

File: src/util/fade_helper.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  m_time += dt_sec;
+
+  if (completed())
+  {
+    m_time = m_total_time;
+    if (m_value)
+      *m_value = m_target;
+    return m_target;
+  }
+  else
```
Comment (tobbi) [inline src/util/fade_helper.cpp:53]: Since you return'ed from the last if, the else statement is redundant. You can reduce the level of nesting here.

Comment (Semphriss) [reply src/util/fade_helper.cpp:53]: I'm not sure to understand... The value computed for each `return` statement is different between the if and the else.

Do you suggest I put a `float return_value;` that I set in both the if and the else, and then have only 1 return statement after the whole if/else?

Comment (Semphriss) [reply src/util/fade_helper.cpp]: Fixed :+1:

**Pull request merged**